api/lib/project-fields-nodes-to-fields-map.js

```diff
@@ -49,7 +49,13 @@
  */
 export function projectFieldsNodesToFieldsMap(state, project, nodes) {
   return Object.entries(project.fields).reduce(
-    (acc, [userInternalFieldName, userFieldName]) => {
+    (acc, [userInternalFieldName, userFieldNameOrConfig]) => {
+      const fieldOptional =
+        typeof userFieldNameOrConfig === "object" &&
+        userFieldNameOrConfig.optional;
+      const userFieldName =
```

Comment:
If `userFieldNameOrConfig.name` is an empty string, `userFieldName` will be set to the field object. It might be worth doing a type check here as well.

Reply:
Good call, thank you!